File: python_solution/problem27.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sat Sep 21 19:34:34 2024

@author: sharminalina
"""

class Problem27:
    def RemoveElement(self, nums: list[int], val: int) -> int:
        
        if not nums:
            return 0

        # Pointer for the position of unique elements
        new_pos = 0
        
        
        for i in range(len(nums)):
           if nums[i] != val:
               nums[new_pos] = nums[i]
               new_pos += 1
                

        # Return the number of unique elements
        return new_pos
    
def main():
    print("Solution of Leet problem No.27: remove all occurrences of given number")
    
    ## Taking input array from the console
    nums_input = input("Enter a array with comma ',' saperated number like 1,2,3: ")
    nums = list(map(int, nums_input.split(',')))
    val = int(input("Enter a value to remove from the list: "))
    solution_prob27 = Problem27()
    result = solution_prob27.RemoveElement(nums, val)
    
    # Printing the length of the array after removing the elements
    print(f"Length after removing the elements: {result}")
    # Printing the array after remove the given item
    print(f"Array after removing the elements: {nums[:result]}")

    
if __name__ == "__main__":
    main()

File: C#Solutions/problem27.cs
using System;
using System.Collections.Generic;

namespace AllProblems
{
    public static class Problem27
    {
        public static void Solve()
        {
            Console.WriteLine("Solution to Problem 27");
            // Read the array from console
            Console.WriteLine("Enter a array of integers (comma-separated):");
            int[] nums = ReadIntArrayFromConsole();

            // Read the integer value to remove the all occurance of that value
            Console.WriteLine("Enter an Integer number to remove from the array");
            int val = int.Parse(Console.ReadLine()); 


            // Find the length of array after remove the duplicate item of sorted array 
            int result = RemoveElement(nums,val);
            Console.WriteLine($"Length of array after remove the given array elements: {result}");
        }

        // Helper method to read an integer array from the console
        private static int[] ReadIntArrayFromConsole()
        {
            string input = Console.ReadLine();
            if (string.IsNullOrWhiteSpace(input))
            {
                return new int[0];
            }

            return input.Split(',')
                        .Select(s => int.Parse(s.Trim()))
                        .ToArray();
        }

        public static int RemoveElement(int[] nums, int val) {

            int new_pos = 0;
            
            for (int i = 0; i < nums.Length; i++){
                if(nums[i] != val){
                    nums[new_pos]= nums[i];
                    new_pos ++;
                    
                }

            }
        
            
            return new_pos;
        }
    }
}